src/kube-runtime/src/init/sshd.sh

```diff
@@ -20,22 +20,69 @@
 SSHD_BIN=/usr/sbin/sshd
 CONFIG_DIR=/usr/local/pai/sshd
 
-if [ -f /usr/sbin/sshd ] ; then
-    if [ -z "$PAI_SSH_PUB_KEY" -o -z "$PAI_CONTAINER_SSH_PORT" ] ; then
-        echo "no private key or port provided" >&2
-    else
-        mkdir -p $CONFIG_DIR
+function prepare_ssh()
+{
+  mkdir /root/.ssh
+  touch /root/.ssh/authorized_keys
+  chmod 600 /root/.ssh/authorized_keys
+
+  mkdir -p /var/run/sshd
+
+# Set sshd config
+  sed -i 's/PermitRootLogin prohibit-password/PermitRootLogin yes/' /etc/ssh/sshd_config
+  sed -i 's/Port.*/Port '$PAI_CONTAINER_SSH_PORT'/' /etc/ssh/sshd_config
+  echo "PermitUserEnvironment yes" >> /etc/ssh/sshd_config
+
+  sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd
+
+  echo "sshd:ALL" >> /etc/hosts.allow
+
+# Set user environment
+  env > /root/.ssh/environment
+}
 
-        mkdir /root/.ssh
-        mkdir -p /var/run/sshd
+function prepare_job_ssh()
+{
+# Job ssh files are mounted to /etc/ssh-secret, can get keys using /etc/ssh-secret/ssh-pub and /etc/ssh-secret/ssh-pri  
+# Please refer to https://kubernetes.io/docs/concepts/configuration/secret/#use-case-pod-with-ssh-keys
+  localPublicKeyPath=/etc/ssh-secret/ssh-publickey
+  localPrivateKeyPath=/etc/ssh-secret/ssh-privatekey
 
-        echo "$PAI_SSH_PUB_KEY" > /root/.ssh/authorized_keys
-        chmod 600 /root/.ssh/authorized_keys
+  if [ -f $localPublicKeyPath ] && [ -f $localPrivateKeyPath ] ; then
+    cat $localPublicKeyPath >> /root/.ssh/authorized_keys
 
-        env > /root/.ssh/environment
-        echo "PermitUserEnvironment yes" >> ${CONFIG_DIR}/config
-        echo "Port ${PAI_CONTAINER_SSH_PORT}" >> ${CONFIG_DIR}/config
-        exec $SSHD_BIN -f ${CONFIG_DIR}/config
+    cp $localPrivateKeyPath /root/.ssh/job_ssh_key
+    chmod 400 /root/.ssh/job_ssh_key
+  else
+    echo "no job ssh keys found" >&2
+  fi
+}
+
+function prepare_user_ssh()
+{
+  if [ -z "$PAI_SSH_PUB_KEY" ] ; then
+    echo "no user ssh key provided" >&2
+  else
+    echo "$PAI_SSH_PUB_KEY" >> /root/.ssh/authorized_keys
+  fi
+}
+
+function start_ssh()
+{
+  printf "%s %s\n" \  
+    "[INFO]" "start ssh service"
+  service ssh restart
+}
+
+
+if [ -f /usr/sbin/sshd ] ; then
+    if [ -z "$PAI_CONTAINER_SSH_PORT" ] ; then
+        echo "no ssh port provided" >&2
+    else
+        prepare_ssh
+        prepare_job_ssh
+        prepare_user_ssh
+        start_ssh
     fi
 else
     echo "no sshd binary found" >&2
```

Comment:
install openssh server for user if there's no sshd in image?

Reply:
Ok